### PR TITLE
Add Design Changelog process

### PR DESCRIPTION
**Pull request by AlicjaSuska**
Adding information about the Design Changelog to our processes in handbook (more context in this [slack thread](https://sourcegraph.slack.com/archives/C014ZCKMCAV/p1662631687355299)).

- I've explained what the Desing Changelog
- When it will be created
- How to use it during and after the project (when revisiting the design decisions)

### CODE REVIEW

File: content/departments/engineering/teams/code-insights/processes.md [hunk 1]
```diff
@@ -81,6 +81,9 @@ In addition to the above iteration planning process, we maintain a parallel desi
...
    1. Past experience has taught us that this causes confusion as to the status of an issue, as it's possible for an issue to from "in progress" on design back to "todo" on engineering.
    1. Often, designs are larger than a single engineering issue and we want to default to engineers breaking up a project into smaller items, rather than reusing a larger and imperfect issue.
+1. During the design phase, the Designer creates and maintains the design changelog that is available as a separate Figma page and it is linked in the main design issue. Design Changelog is created to keep track of the key or pivotal design decisions for a given project as well as the reasoning behind them.
+   1. Design Changelog is mostly useful for big projects. Smaller issues may not require creating the backlog.
+   1. When re-opening the discussion about the past design decision, please read the design changelog first to make sure that your point wasn't addressed earlier and that new information has occured.
```
Comment (felixfbecker) [inline content/departments/engineering/teams/code-insights/processes.md]: Mostly Grammarly fixes, add missing articles, consistent casing, slight wording tweaks:
```suggestion
1. During the design phase, the Designer creates and maintains a design changelog that is available as a separate Figma page and is linked from the main design issue. The design changelog is created to keep track of the key or pivotal design decisions for a given project as well as the reasoning behind them.
   1. The design changelog is primarily useful for big projects. Smaller issues may not require creating the changelog.
   1. When re-opening the discussion about the past design decision, please read the design changelog first to make sure that your point wasn't addressed earlier and that new information came to light that warrants reconsidering the decision.
```

**Review by felixfbecker — APPROVED** [review]
Thank you for adding!
Would like to get review from the rest of the team too, especially @sourcegraph/code-insights-frontend

**Review by unclejustin — APPROVED** [review]

**Review by Joelkw — APPROVED** [review]
Thank you for creating this!

**Review by vovakulikov — APPROVED** [review]

**Comment by felixfbecker** [comment]
@AlicjaSuska could you push this across the finish line? 🏁  It was approved by everyone and I made a suggestion above.

**Pull request merged**